test/Faker/GeneratorTest.php

```diff
@@ -157,6 +159,131 @@ public function testSeed()
         $this->faker->seed('10');
         self::assertTrue(true, 'seeding with a non int value doesn\'t throw an exception');
     }
+
+    public function testUniqueReturnsUniqueGenerator(): void
+    {
+        $generator = new Generator();
+
+        $generator->addProvider(new Provider\Base($generator));
+
+        $uniqueGenerator = $generator->unique();
+
+        self::assertInstanceOf(UniqueGenerator::class, $uniqueGenerator);
+    }
+
+    public function testUniqueReturnsSameUniqueGeneratorWhenUsedWithoutArguments(): void
+    {
+        $generator = new Generator();
+
+        $generator->addProvider(new Provider\Base($generator));
+
+        $uniqueGenerator = $generator->unique();
+
+        self::assertSame($uniqueGenerator, $generator->unique());
+    }
+
+    public function testUniqueReturnsSameUniqueGeneratorWhenResetIsFalse(): void
+    {
+        $generator = new Generator();
+
+        $generator->addProvider(new Provider\Base($generator));
+
+        $uniqueGenerator = $generator->unique();
+
+        self::assertSame($uniqueGenerator, $generator->unique(false));
+    }
+
+    public function testUniqueReturnsDifferentUniqueGeneratorWhenResetIsTrue(): void
+    {
+        $generator = new Generator();
+
+        $generator->addProvider(new Provider\Base($generator));
+
+        $uniqueGenerator = $generator->unique();
+
+        self::assertNotSame($uniqueGenerator, $generator->unique(true));
+    }
+
+    public function testUniqueReturnsUniqueGeneratorThatGeneratesUniqueValues(): void
+    {
+        $words = [
+            'foo',
+            'bar',
+            'baz',
+        ];
+
+        $generator = new Generator();
+
+        $generator->addProvider(new Provider\Base($generator));
+        $generator->addProvider(new class(...$words) {
+            private $words;
+
+            public function __construct(string ...$words)
```

Comment:
Why the array deconstruction and not just pass the word array?

Reply:
@bramceulemans 

Because when passing an array, it would be necessary to write more code to ensure that every element of the arrat is a `string`:

```php
/**
 * @param array<int, string> $words
 */
public function __construct(array $words)
{
    $this->words = array_values(array_map(static function (string $word): string {
        return $word;
    }, $words));
}
```

or

```php
/**
 * @param array<int, string> $words
 * 
 * @throws \InvalidArgumentException
 */
public function __construct(array $words)
{
    $notString = array_filter(static function ($word): bool {
        return !is_string($word);
    }, $words);

    if ([] !== $notString) {
        throw new \InvalidArgumentException('Every word needs to be a string');
    }

    $this->words = array_values($words);
}
```

Using variadics here allows me not to write this code.